### PR TITLE
Flex merge feeds

### PR DESCRIPTION
**Pull request by philip-cline**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Flex updates to support flex feed merging in datatools. This PR was reopened by Phil. All feedback should be addressed to @br648 .

### CODE REVIEW

**Comment by philip-cline** [comment]
This PR has been reopened in the forked repository

File: src/main/java/com/conveyal/gtfs/model/StopTime.java [hunk 1]
```diff
@@ -240,44 +241,49 @@ private static boolean flexColumnExist(Connection connection, String tablePrefix
...
+                    stopTime.mean_duration_offset  = getDoubleValue(resultSet.getString(11));
+                    stopTime.safe_duration_factor  = getDoubleValue(resultSet.getString(12));
+                    stopTime.safe_duration_offset  = getDoubleValue(resultSet.getString(13));
+                    stopTimes.add(stopTime);
+                }
+            }
+            return stopTimes;
+        } finally {
+            // Close transaction finally.
+            if (connection != null) DbUtils.closeQuietly(connection);
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/gtfs/model/StopTime.java]: Put the result of `dataSource.getConnection()` as a "resource" or close within the `finally` statement of the `try` block that calls this method, and remove the `finally` block (and probably the entire corresponding `try` statement) from this method.

File: src/main/java/com/conveyal/gtfs/util/GeoJsonUtil.java [hunk 2]
```diff
@@ -295,6 +291,42 @@ public static CsvReader getCsvReaderFromGeoJson(
         return new CsvReader(new StringReader(csvContent.toString()));
     }
 
+    /**
+     * Extract the locations from Geo Json.
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/gtfs/util/GeoJsonUtil.java]: Fix the GeoJSON spelling (see https://geojson.org/) and update the other comments too.
```suggestion
     * Extract the locations from GeoJSON.
```

File: src/main/java/com/conveyal/gtfs/util/GeoJsonUtil.java [hunk 3]
```diff
@@ -295,6 +291,42 @@ public static CsvReader getCsvReaderFromGeoJson(
...
+        return GeoJsonUtil.unpackLocationShapes(features, errors);
+    }
+
+    /**
+     * Extract the Geo Json features from file.
+     */
+    private static FeatureCollection getFeaturesFromGeoJson(ZipFile zipFile, ZipEntry entry, List<String> errors) {
+        FeatureCollection features = GeoJsonUtil.getFeatureCollection(zipFile, entry);
+        if (features == null || features.numFeatures() == 0) {
+            String message = "Unable to extract GeoJson features (or none are available) from " + entry.getName();
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/gtfs/util/GeoJsonUtil.java]: Fix the "GeoJSON" spelling.

File: src/test/java/com/conveyal/gtfs/GTFSTest.java [hunk 4]
```diff
@@ -1406,7 +1446,6 @@ private void assertThatPersistenceExpectationRecordWasFound(
             "stop_areas",
             new RecordExpectation[]{
                 new RecordExpectation("area_id", "area1"),
-                new RecordExpectation("stop_id", "123")
```
Comment (binh-dam-ibigroup) [inline src/test/java/com/conveyal/gtfs/GTFSTest.java:1409]: Also delete trailing comma on previous line

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Please cleanup some `try` blocks and spelling, please.

File: src/main/java/com/conveyal/gtfs/util/GeoJsonUtil.java (hunk 3 above)
Comment (br648) [reply src/main/java/com/conveyal/gtfs/util/GeoJsonUtil.java]: @binh-dam-ibigroup Do you think I should update all classes and methods as well?

Comment (binh-dam-ibigroup) [reply src/main/java/com/conveyal/gtfs/util/GeoJsonUtil.java]: Maybe in another PR that could also be against upstream?

File: src/main/java/com/conveyal/gtfs/util/GeoJsonUtil.java [hunk 5]
```diff
@@ -296,7 +292,43 @@ public static CsvReader getCsvReaderFromGeoJson(
     }
 
     /**
-     * Convert {@link Location} and {@link LocationShape} lists to a serialized String conforming to the GeoJson
+     * Extract the locations from GeoJSON.
+     */
+    public static List<Location> getLocationsFromGeoJson(ZipFile zipFile, ZipEntry entry, List<String> errors) {
+        FeatureCollection features = getFeaturesFromGeoJson(zipFile, entry, errors);
+        if (features == null) {
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/gtfs/util/GeoJsonUtil.java]: Might be able to use a ternary here and in the next function.

**Review by binh-dam-ibigroup — APPROVED** [review]

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Are we sure this PR is good to go? Since stop areas has been re-renamed

**Comment by br648** [comment]
> Are we sure this PR is good to go? Since stop areas has been re-renamed

@miles-grant-ibigroup The thinking is to get this merged albeit with the wrong naming. Once this aspect of the flex spec is set-in-stone, then update across the board.

**Review by miles-grant-ibigroup — APPROVED** [review]
Gotcha. Great fixes thanks!

**Pull request merged**